storage/src/error.rs

```diff
@@ -2,23 +2,52 @@
 
 use witnet_util as util;
 
-/// Enumerates all the error types generated by the Storage module.
-#[derive(Debug)]
-pub enum StorageErrors {
+use failure::Fail;
+
+/// Storage Error caused be `C`.
+#[derive(Debug, Fail)]
+pub enum StorageError {
     /// Error while trying to open a connection to the storage backend.
-    ConnectionError,
+    #[fail(
+        display = "connection error: at \"{}\", rocksdb_msg {}",
+        path,
+        msg
+    )]
+    Connection {
+        /// Connection address
+        path: String,
```

Comment:
Ok, I get it. You used a `String` in `InMemoryStorage::new()` because of this.

But the thing is that `Storage` and therefore `StorageError` must remain generic as we don't want to limit different implementors to only accept `String` as connection info. I mean, this argument could sometimes be a path, some times an URL and some times a data structure with user, password, etc. 

What about this?
```rust
pub enum StorageError<T: Debug> {
    #[fail(
        display = "connection error: with data \"{:?}\" and msg \"{}\"",
        path,
        msg
    )]
    Connection {
        data: T,
        msg: String
    }
}
```
For the same reason, we shouldn't mention here `rocksdb_msg` but simply `msg`.

Generally speaking, we want to decouple our architecture from specific dependencies, so that we can easily swap them in the future (or even let the users use whatever storage backend they want to use).